### PR TITLE
Ports Vox to sunset --- Still working on refactoring sprites

### PR DESCRIPTION
**Pull request by SunsetStationuser**
:cl: Bksparkie1 -- Code thanks to Ziiro and Paradise
add: Vox species
/:cl:

[why]: # Many people requested it was ported over.

### CODE REVIEW

**Comment by alexkar598** [comment]
This will be he'll to review

**Comment by alexkar598** [comment]
Hell*

**Comment by miniusAreas** [comment]
next stop - corvids and IPC.

**Comment by Wilchenx** [comment]
next stop - tajara

File: code/datums/emotes.dm [hunk 1]
```diff
@@ -21,6 +21,7 @@
 	var/list/mob_type_ignore_stat_typecache
 	var/stat_allowed = CONSCIOUS
 	var/static/list/emote_list = list()
+	var/message_vox = "" // Message to display if the user is a vox sunset
```
Comment (ghost) [inline code/datums/emotes.dm]: Comment is formatted wrong, it should be `// sunset -- Message to display if the user is a vox sunset`

File: code/modules/cargo/packs.dm [hunk 2]
```diff
@@ -121,6 +121,9 @@
 					/obj/item/tank/internals/emergency_oxygen,
 					/obj/item/tank/internals/emergency_oxygen,
 					/obj/item/tank/internals/emergency_oxygen,
+					/obj/item/tank/internals/emergency_oxygen/vox,	// sunset vox emergency oxygen tank
```
Comment (ghost) [inline code/modules/cargo/packs.dm]: `// sunset -- `

Comment (ghost) [reply code/modules/cargo/packs.dm]: Or you could just do

```
// sunset start
/obj/item/tank/internals/emergency_oxygen/vox,
/obj/item/tank/internals/emergency_oxygen/vox,
/obj/item/tank/internals/emergency_oxygen/vox,
// sunset end
```

File: code/modules/client/preferences.dm [hunk 3]
```diff
@@ -386,6 +386,59 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 					dat += "</td>"
 					mutant_category = 0
 
+			if("vox_body" in pref_species.mutant_bodyparts) //sunset vox bodyparts
```
Comment (ghost) [inline code/modules/client/preferences.dm]: Make this a hook proc. Like

```dat += sunset_preferences()```

File: code/modules/client/preferences.dm [hunk 4]
```diff
@@ -1285,6 +1338,36 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 					if(new_moth_wings)
 						features["moth_wings"] = new_moth_wings
 
+				if("vox_body") //sunset vox included
```
Comment (ghost) [inline code/modules/client/preferences.dm]: Hook!

File: code/modules/client/preferences_savefile.dm [hunk 5]
```diff
@@ -236,6 +236,11 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 	S["feature_lizard_body_markings"]	>> features["body_markings"]
 	S["feature_lizard_legs"]			>> features["legs"]
 	S["feature_moth_wings"]				>> features["moth_wings"]
+	S["feature_vox_body"]				>> features["vox_body"] //sunset vox body parts
```
Comment (ghost) [inline code/modules/client/preferences_savefile.dm]: Can be made into a hook proc.

File: code/modules/client/preferences_savefile.dm [hunk 6]
```diff
@@ -378,6 +390,11 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 	WRITE_FILE(S["feature_lizard_body_markings"]	, features["body_markings"])
 	WRITE_FILE(S["feature_lizard_legs"]			, features["legs"])
 	WRITE_FILE(S["feature_moth_wings"]			, features["moth_wings"])
+	WRITE_FILE(S["feature_vox_body"]				, features["vox_body"]) //sunset vox body parts
```
Comment (ghost) [inline code/modules/client/preferences_savefile.dm]: Can be made into a hook proc.

File: code/modules/jobs/job_types/medical.dm [hunk 7]
```diff
@@ -194,7 +194,7 @@ Virologist
 	belt = /obj/item/pda/viro
 	ears = /obj/item/radio/headset/headset_med
 	uniform = /obj/item/clothing/under/rank/virologist
-	mask = /obj/item/clothing/mask/surgical
+	backpack_contents = list(/obj/item/clothing/mask/surgical=1) //sunset vox mask
```
Comment (ghost) [inline code/modules/jobs/job_types/medical.dm]: You can override this one var in `sunset/code/modules/jobs/job_types/medical.dm`

File: code/modules/jobs/job_types/security.dm [hunk 8]
```diff
@@ -161,22 +161,22 @@ Detective
 	l_pocket = /obj/item/toy/crayon/white
 	r_pocket = /obj/item/lighter
 	backpack_contents = list(/obj/item/storage/box/evidence=1,\
```
Comment (ghost) [inline code/modules/jobs/job_types/security.dm]: Can be overriden externally

File: code/modules/jobs/job_types/security.dm [hunk 9]
```diff
@@ -161,22 +161,22 @@ Detective
...
 	implants = list(/obj/item/implant/mindshield)
 
 	chameleon_extras = list(/obj/item/gun/ballistic/revolver/detective, /obj/item/clothing/glasses/sunglasses)
 
-/datum/outfit/job/detective/post_equip(mob/living/carbon/human/H, visualsOnly = FALSE)
-	..()
-	var/obj/item/clothing/mask/cigarette/cig = H.wear_mask
-	if(istype(cig)) //Some species specfic changes can mess this up (plasmamen)
-		cig.light("")
+// /datum/outfit/job/detective/post_equip(mob/living/carbon/human/H, visualsOnly = FALSE) Sunset vox merge
```
Comment (ghost) [inline code/modules/jobs/job_types/security.dm]: Multiline comments!

```
/*
...
*/
```

Or simply add a check to see if a mask is already there.

File: code/modules/mob/living/carbon/human/species.dm [hunk 10]
```diff
@@ -566,6 +566,26 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 			bodyparts_to_add -= "wings_open"
 		else if ("wings" in mutant_bodyparts)
 			bodyparts_to_add -= "wings_open"
+				//vox sunsetstation
+	if("vox_quills" in mutant_bodyparts)
```
Comment (ghost) [inline code/modules/mob/living/carbon/human/species.dm]: Hook proc!

File: code/modules/mob/living/carbon/human/species.dm [hunk 11]
```diff
@@ -634,6 +654,20 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 					S = GLOB.moth_wings_list[H.dna.features["moth_wings"]]
 				if("caps")
 					S = GLOB.caps_list[H.dna.features["caps"]]
+				if("vox_body") //sunset vox bodyparts
```
Comment (ghost) [inline code/modules/mob/living/carbon/human/species.dm]: Hook proc!

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 12]
```diff
@@ -129,11 +129,20 @@ There are several things that need to be remembered:
 		if(dna && dna.species.sexes)
 			var/G = (gender == FEMALE) ? "f" : "m"
 			if(G == "f" && U.fitted != NO_FEMALE_UNIFORM)
-				uniform_overlay = U.build_worn_icon(state = "[t_color]", default_layer = UNIFORM_LAYER, default_icon_file = 'icons/mob/uniform.dmi', isinhands = FALSE, femaleuniform = U.fitted)
+				if(U.icon_override)//Done for vox and future species Sunset
```
Comment (ghost) [inline code/modules/mob/living/carbon/human/update_icons.dm]: ```// sunset start```
and ```// sunset end```

File: code/modules/surgery/organs/heart.dm [hunk 13]
```diff
@@ -178,3 +178,10 @@
 		owner.heal_overall_damage(15, 15)
 		if(owner.reagents.get_reagent_amount("ephedrine") < 20)
 			owner.reagents.add_reagent("ephedrine", 10)
+
+/obj/item/organ/heart/vox
```
Comment (ghost) [inline code/modules/surgery/organs/heart.dm]: `sunset/code/modules/surgery/organs/heart.dm`

File: code/modules/surgery/organs/eyes.dm [hunk 14]
```diff
@@ -338,3 +338,12 @@
 	name = "moth eyes"
 	desc = "These eyes seem to have increased sensitivity to bright light, with no improvement to low light vision."
 	flash_protect = -1
+
```
Comment (ghost) [inline code/modules/surgery/organs/eyes.dm]: `sunset/code/modules/surgery/organs/eyes.dm`

File: code/modules/surgery/organs/liver.dm [hunk 15]
```diff
@@ -88,3 +88,9 @@
 			damage+=100
 		if(2)
 			damage+=50
+
+/obj/item/organ/liver/vox
```
Comment (ghost) [inline code/modules/surgery/organs/liver.dm]: `sunset/ code/modules/surgery/organs/liver.dm`

**Review by ghost — CHANGES_REQUESTED** [review]
I'm not even halfway through, this is NOT modularized

**Comment by ghost** [comment]
tajara will be insta closed

File: sunset/code/modules/hydroponics/beekeeping/beekeeper_suit.dm [hunk 16]
```diff
@@ -0,0 +1,6 @@
+/obj/item/clothing/head/beekeeper_head
+	species_restricted = list("exclude","Vox Outcast")
```
Comment (Ziiro) [inline sunset/code/modules/hydroponics/beekeeping/beekeeper_suit.dm:2]: https://github.com/OracleStation/OracleStation/pull/712

File: sunset/code/modules/clothing/suits/cloaks.dm [hunk 17]
```diff
@@ -0,0 +1,10 @@
+/obj/item/clothing/neck/cloak
+	species_restricted = list("exclude","Vox Outcast")
+
+
+/obj/item/clothing/suit/hooded/cloak/goliath
+	species_restricted = list("exclude","Vox Outcast")
+
+
+/obj/item/clothing/suit/hooded/cloak/drake
```
Comment (Ziiro) [inline sunset/code/modules/clothing/suits/cloaks.dm:9]: https://github.com/OracleStation/OracleStation/pull/919

**Comment by Ziiro** [comment]
also might want to check this PR for things I fixed:
https://github.com/OracleStation/OracleStation/pull/689

File: code/modules/surgery/organs/liver.dm (hunk 15 above)
Comment (SunsetStationuser) [reply code/modules/surgery/organs/liver.dm]: already did, was having troubles, moved them only temporarily

File: code/modules/client/preferences.dm [hunk 18]
```diff
@@ -443,6 +443,7 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 			if(mutant_category)
 				dat += "</td>"
 				mutant_category = 0
+			dat += vox_preference_data(user)
```
Comment (ghost) [inline code/modules/client/preferences.dm]: ```// sunset -- adds vox preferences```

File: code/modules/client/preferences.dm [hunk 19]
```diff
@@ -1346,6 +1347,9 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 					if(pickedPDAColor)
 						pda_color = pickedPDAColor
 
+				else
+					process_vox(user)
```
Comment (ghost) [inline code/modules/client/preferences.dm]: ```// sunset -- adds vox preferences```

File: code/modules/client/preferences_savefile.dm [hunk 20]
```diff
@@ -236,6 +236,8 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 	S["feature_lizard_body_markings"]	>> features["body_markings"]
 	S["feature_lizard_legs"]			>> features["legs"]
 	S["feature_moth_wings"]				>> features["moth_wings"]
+	load_character_vox(S)
```
Comment (ghost) [inline code/modules/client/preferences_savefile.dm]: ```// sunset -- adds vox features```

File: code/modules/client/preferences_savefile.dm [hunk 21]
```diff
@@ -320,6 +322,7 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 	features["body_markings"] 	= sanitize_inlist(features["body_markings"], GLOB.body_markings_list)
 	features["feature_lizard_legs"]	= sanitize_inlist(features["legs"], GLOB.legs_list, "Normal Legs")
 	features["moth_wings"] 	= sanitize_inlist(features["moth_wings"], GLOB.moth_wings_list, "Plain")
+	load_character_vox_features()
```
Comment (ghost) [inline code/modules/client/preferences_savefile.dm]: ```// sunset -- adds vox features```

File: code/modules/client/preferences_savefile.dm [hunk 22]
```diff
@@ -378,6 +381,7 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 	WRITE_FILE(S["feature_lizard_body_markings"]	, features["body_markings"])
 	WRITE_FILE(S["feature_lizard_legs"]			, features["legs"])
 	WRITE_FILE(S["feature_moth_wings"]			, features["moth_wings"])
+	save_character_vox(S)
```
Comment (ghost) [inline code/modules/client/preferences_savefile.dm]: ```// sunset -- adds vox features```

File: code/modules/mob/living/carbon/human/species.dm [hunk 23]
```diff
@@ -566,6 +566,8 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 			bodyparts_to_add -= "wings_open"
 		else if ("wings" in mutant_bodyparts)
 			bodyparts_to_add -= "wings_open"
+	else
+		bodyparts_to_add -= handle_vox_bodyparts(H)
```
Comment (ghost) [inline code/modules/mob/living/carbon/human/species.dm]: ```// sunset -- adds vox stuff```

File: code/modules/mob/living/carbon/human/species.dm [hunk 24]
```diff
@@ -634,6 +636,8 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 					S = GLOB.moth_wings_list[H.dna.features["moth_wings"]]
 				if("caps")
 					S = GLOB.caps_list[H.dna.features["caps"]]
+				else
+					handle_vox_bodyparts_features(H, S)
```
Comment (ghost) [inline code/modules/mob/living/carbon/human/species.dm]: ```// sunset -- adds vox stuff```

File: code/modules/surgery/organs/tongue.dm [hunk 25]
```diff
@@ -17,6 +17,7 @@
 		/datum/language/beachbum,
 		/datum/language/ratvar,
 		/datum/language/aphasia,
+		/datum/language/voxpidgin,//sunsetstation adds vox language to tongue
```
Comment (ghost) [inline code/modules/surgery/organs/tongue.dm]: ```// sunset -- adds vox language to tongue```

File: sunset/code/__HELPERS/mobs.dm [hunk 26]
```diff
@@ -0,0 +1,38 @@
+random_features()
```
Comment (ghost) [inline sunset/code/__HELPERS/mobs.dm:1]: /

**Review by ghost — CHANGES_REQUESTED** [review]
Looks mostly good to me, but I kinda rushed it because i'm tired.

DON'T FUCKING EDIT /tg/ DMIs!!!

**Review by ghost — COMMENTED** [review]
Test it,

File: code/modules/surgery/organs/lungs.dm [hunk 27]
```diff
@@ -24,6 +24,7 @@
 	var/oxy_breath_dam_min = MIN_TOXIC_GAS_DAMAGE
 	var/oxy_breath_dam_max = MAX_TOXIC_GAS_DAMAGE
 	var/oxy_damage_type = OXY
+	var/oxy_damage_multiplier = 1
```
[the remark this answers is not in the document]
Comment (SunsetStationuser) [reply code/modules/surgery/organs/lungs.dm:27]: planning on making it affect humans and everything else in the future

**Comment by Ziiro** [comment]
oh shit just like that huh

**Pull request merged**